spring-statemachine-core/src/main/java/org/springframework/statemachine/config/model/TransitionData.java

```diff
@@ -108,6 +160,7 @@ public TransitionData(S source, S target, S state, E event, Long period, Integer
 		this.guard = guard;
 		this.kind = kind;
 		this.securityRule = securityRule;
+		this.name = name;
```

Comment:
You could do null check here as set it to empty, then from other constructors pass in null.